src/program/lwaftr/tests/subcommands/config_test.py

```diff
@@ -0,0 +1,228 @@
+            # Implicit string concatenation, do not add commas.
+            '/softwire-config/binding/br/br-instances/'
+            'br-instance[id=1]/binding-table/binding-entry'
+            '[binding-ipv6info=127:22:33:44:55:66:77:128]/binding-ipv4-addr'))
+        output = self.run_cmd(cmd_args)
+        self.assertEqual(output.strip(), b'178.79.150.15',
+            '\n'.join(('OUTPUT', str(output, ENC))))
+
+
+class TestConfigListen(BaseTestCase):
+    """
+    Test it can listen, send a command and get a response. Only test the
+    socket method of communicating with the listen command, due to the
+    difficulties of testing interactive scripts.
+    """
+
+    daemon_args = DAEMON_ARGS
+    wait_for_daemon_startup = True
+
+    listen_args = (str(SNABB_CMD), 'config', 'listen',
+        '--socket', SOCKET_PATH, DAEMON_PROC_NAME)
+
+    def test_listen(self):
+        # Start the listen command with a socket.
+        listen_daemon = Popen(self.listen_args, stdout=PIPE, stderr=PIPE)
+        # Wait a short while for the socket to be created.
+        time.sleep(1)
+        # Send command to and receive response from the listen command.
+        # (Implicit string concatenation, no summing needed.)
+        get_cmd = (b'{ "id": "0", "verb": "get",'
+            b' "path": "/routes/route[addr=1.2.3.4]/port" }\n')
+        sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        try:
+            sock.connect(SOCKET_PATH)
+            sock.sendall(get_cmd)
+            resp = str(sock.recv(200), ENC)
+        finally:
+            sock.close()
+        status = json.loads(resp)['status']
+        self.assertEqual(status, 'ok')
+        # Terminate the listen command.
+        listen_daemon.terminate()
+        ret_code = listen_daemon.wait()
+        if ret_code not in (0, -SIGTERM):
+            print('Error terminating daemon:', listen_daemon.args)
+            print('Exit code:', ret_code)
+            print('STDOUT\n', str(listen_daemon.stdout.read(), ENC))
+            print('STDERR\n', str(listen_daemon.stderr.read(), ENC))
+        listen_daemon.stdout.close()
+        listen_daemon.stderr.close()
+        os.unlink(SOCKET_PATH)
+
+
+class TestConfigMisc(BaseTestCase):
+
+    daemon_args = DAEMON_ARGS
+    wait_for_daemon_startup = True
+
+    def get_cmd_args(self, action):
+        cmd_args = list((str(SNABB_CMD), 'config', 'XXX', DAEMON_PROC_NAME))
+        cmd_args[2] = action
+        return cmd_args
+
+    def test_add(self):
+        """
+        Add a softwire section, get it back and check all the values.
+        """
+        # External IPv4.
+        add_args = self.get_cmd_args('add')
+        add_args.extend(('/softwire-config/binding-table/softwire',
+            '{ ipv4 1.2.3.4; psid 72; b4-ipv6 ::1; br 1; }'))
+        self.run_cmd(add_args)
+        get_args = self.get_cmd_args('get')
+        get_args.append('/softwire-config/binding-table/softwire[ipv4=1.2.3.4][psid=72]')
```

Comment:
I usually try to keep lines under 80 chars, and always under 90. It looks like there's agreement for being strict at 80, so I'll shorten lines all over the place.